lib/widgets/profile.dart

```diff
@@ -0,0 +1,334 @@
+          padding: const EdgeInsets.all(4),
+          child: ProfileDataTableLink(url: entry.text, label: entry.text));
+      case CustomProfileFieldType.choice:
+        // TODO(server): This isn't really documented.  But see chat thread:
+        //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1383005
+        final Map<String, dynamic> fieldData = jsonDecode(entry.fieldData);
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(fieldData[entry.text]["text"]));
+      case CustomProfileFieldType.externalAccount:
+        // TODO(server): This is undocumented.  See chat thread:
+        //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/external.20account.20custom.20profile.20fields/near/1387213
+        final Map<String, dynamic> fieldData = jsonDecode(entry.fieldData);
+        String? urlPattern = fieldData["url_pattern"];
+        if (urlPattern == null) {
+          final RealmDefaultExternalAccount? realmDefaultExternalAccount = store.realmDefaultExternalAccounts[fieldData["subtype"]];
+          if (realmDefaultExternalAccount != null) {
+            urlPattern = realmDefaultExternalAccount.urlPattern;
+          }
+        }
+        if (urlPattern != null) {
+          final url = urlPattern.replaceFirst("%(username)s", entry.text);
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: ProfileDataTableLink(url: url, label: entry.text));
+        } else {
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: Text(entry.text));
+        }
+      case CustomProfileFieldType.user:
+        // TODO(server): This is completely undocumented.  The key to
+        //   reverse-engineering it was:
+        //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+        final List<dynamic> userIds = jsonDecode(entry.text);
+        if (userIds.isEmpty) {
+          return const SizedBox.shrink();
+        }
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: userIds.whereType<num>()
+            .map((userId) => store.users[userId])
+            .whereType<User>()
+            .map((user) => ProfileDataTableUser(user: user)).toList());
+      case CustomProfileFieldType.date:
+      // TODO(server): The value's format is undocumented, but empirically
+      //   it's a date in ISO format, like 2000-01-01.
+      // That's readable as is, but:
+      // TODO format this date using user's locale.
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+      case CustomProfileFieldType.unknown:
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
```

Comment:
If the type is one we don't know about, this will cheerfully carry on and show the raw contents (which are likely some kind of JSON blob) to the user as if the other user had typed that in as plain text. That's potentially pretty odd-looking.

Looks like the zulip-mobile implementation handles this case by skipping that custom profile field entirely, which I think is pretty reasonable. The most thorough thing would I guess be to instead explicitly say that it's there and that this app doesn't support it.